vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/ScrollContainer.java

```diff
@@ -0,0 +1,174 @@
+     * Scroll direction defaults to {@link ScrollDirection#BOTH} and can be set
+     * with {@link #setScrollDirection(ScrollDirection)}.
+     * 
+     * @param content
+     *            the content of this container
+     * @see #setContent(Component)
+     */
+    public ScrollContainer(Component content) {
+        this();
+        setContent(content);
+    }
+
+    /**
+     * Convenience constructor to create a container with the given content and
+     * scroll direction.
+     *
+     * @param content
+     *            the content of this container
+     * @param scrollDirection
+     *            scroll direction that the container will have
+     */
+    public ScrollContainer(Component content, ScrollDirection scrollDirection) {
+        this();
+        setContent(content);
+        setScrollDirection(scrollDirection);
+    }
+
+    /**
+     * Convenience constructor to create an empty container with the given
+     * scroll direction. Content can be set with {@link #setContent(Component)}
+     *
+     * @param scrollDirection
+     *            scroll direction that the container will have
+     */
+    public ScrollContainer(ScrollDirection scrollDirection) {
+        this();
+        setScrollDirection(scrollDirection);
+    }
+
+    /**
+     * Sets the content of this container.
+     *
+     * The content must always be set, either with a constructor parameter or by
+     * calling this method.
+     * 
+     * @param content
+     *            a component to use as content
+     */
+    public void setContent(Component content) {
+        if (this.content != null) {
+            this.content.getElement().removeFromParent();
+        }
+        this.content = content;
+        this.getElement().appendChild(content.getElement());
+    }
+
+    /**
+     * Gets the content of this container.
+     *
+     * @return the component used as content
+     */
+    public Component getContent() {
+        return content;
+    }
+
+    /**
+     * Sets the scroll direction for this container. Defaults to
+     * {@link ScrollDirection#BOTH}.
+     *
+     * @param scrollDirection
+     *            {@link ScrollDirection#BOTH} to enable both vertical and
+     *            horizontal scrollbars. {@link ScrollDirection#HORIZONTAL} to
+     *            enable only horizontal scrollbars.
+     *            {@link ScrollDirection#VERTICAL} to enable only vertical
+     *            scrollbars. {@link ScrollDirection#NONE} to disable both
+     *            vertical and horizontal scrollbars.
+     */
+    public void setScrollDirection(ScrollDirection scrollDirection) {
+        requireNonNull(scrollDirection, "Scroll direction must not be null");
+        String webComponentValue = scrollDirection.toWebComponentValue();
+        if (webComponentValue == null) {
+            getElement().removeAttribute(SCROLL_DIRECTION);
+        } else {
+            getElement().setAttribute(SCROLL_DIRECTION, webComponentValue);
+        }
```

Comment:
No need to set the property trough Polymer's attribute deserialization. `getElement().setProperty(SCROLL_DIRECTION, webComponentValue)` should be enough?

Reply:
Same with the getter; should be enough to just return the property value

Reply:
There was no property initially, it's an internal implementation detail, but makes things simpler, done 👍